scripts/verify.ts

```diff
@@ -5,41 +5,89 @@ import fs from 'fs';
 import hre, { deployments } from 'hardhat';
 import path from 'path';
 
-async function verify() {
+export interface ContractToVerify {
+    name: string;
+    address: string;
+    constructorArguments?: any[];
+}
+
+/**
+ * Gets contracts to verify from deployments directory.
+ */
+async function getContractsFromDeployments(): Promise<ContractToVerify[]> {
     const jsonExtension = '.json';
     const contractNames = fs
         .readdirSync(path.resolve(__dirname, `../deployments/${hre.network.name}`))
         .filter((file) => file.endsWith(jsonExtension))
         .map((filePath) => filePath.replace(jsonExtension, ''));
 
-    console.log(`Contracts to verify: ${contractNames}`);
+    if (contractNames.length === 0) {
+        console.log(`\nNo contracts to verify on network: ${hre.network.name}`);
+        return [];
+    }
 
+    const contracts: ContractToVerify[] = [];
     for (const contractName of contractNames) {
+        const deployment = await deployments.get(contractName);
+        contracts.push({
+            name: contractName,
+            address: deployment.address,
+            constructorArguments: deployment.args || [],
+        });
+    }
+    return contracts;
+}
+
+/**
+ * Verifies contracts on block explorer (e.g., Etherscan, Arbiscan).
+ * Can verify either specific contracts or all contracts from deployments directory.
+ *
+ * @param contracts - Optional array of specific contracts to verify. If not provided,
+ *                    will verify all contracts from the deployments/{network} directory.
+ */
+async function verify(contracts?: ContractToVerify[]): Promise<void> {
```

Comment:
Let's add a function
```
function tryVerify() { try {verify()} catch(e) {} }
```
and call it from deployment so that it does not break it and artifacts are correctly committed.